### PR TITLE
Model Editor Checkbox Fix

### PR DESCRIPTION
**Pull request by robelgeda**
Model editor's `Fixed` column crashes when the checkbox is clicked. This PR fixes this bug. I also added "Plot Units" to every unit display in the model editor param box. Lastly, I made sure that the model editor redraws a plot only if the selected `plot_data_item` is of type `ModelDataItem`.

### CODE REVIEW

File: specviz/plugins/model_editor/items.py [hunk 1]
```diff
@@ -22,6 +22,11 @@ def flux(self):
 
         return self.data(self.DataRole).flux
 
+    @property
+    def spectrum(self):
+        flux = self.flux  # Update the flux
+        return super().spectrum
+
```
Comment (drdavella) [inline specviz/plugins/model_editor/items.py]: Is this change related to the rest of the PR?

File: specviz/plugins/model_editor/model_editor.py [hunk 2]
```diff
@@ -192,17 +192,19 @@ def _redraw_model(self):
...
             # If the item has a parent, then we know that the parameter
             # value has changed. Note that the internal stored data has not
             # been truncated at all, only the displayed text value. All fitting
             # uses the full, un-truncated data value.
-            item.setData(float(item.text()), Qt.UserRole + 1)
-            # item.setText("{:.5g}".format(float(item.text()))) # dont change user input
-            item.setText(item.text())
+            if item.column() == 1:
+                item.setData(float(item.text()), Qt.UserRole + 1)
+                # item.setText("{:.5g}".format(float(item.text()))) # dont change user input
```
Comment (drdavella) [inline specviz/plugins/model_editor/model_editor.py]: Should remove this comment if it's no longer useful.

File: specviz/plugins/model_editor/models.py [hunk 3]
```diff
@@ -77,7 +77,8 @@ def add_model(self, model):
             param_value.setData(parameter.value, Qt.UserRole + 1)
 
             # Store the unit information
-            param_unit = QStandardItem("{}".format(parameter.unit))
+            # param_unit = QStandardItem("{}".format(parameter.unit))
```
Comment (drdavella) [inline specviz/plugins/model_editor/models.py]: Should remove this comment if it's no longer useful.

Comment (robelgeda) [reply specviz/plugins/model_editor/models.py]: We need the commented out lines when units are enabled. They are there because those lines are not final.

File: specviz/plugins/model_editor/model_editor.py (hunk 2 above)
Comment (robelgeda) [reply specviz/plugins/model_editor/model_editor.py]: This line is not final. There needs to be a discussion on how to display the fitted params.

File: specviz/plugins/model_editor/items.py (hunk 1 above)
Comment (robelgeda) [reply specviz/plugins/model_editor/items.py]: This is related to the 

> I made sure that the model editor redraws a plot only if the selected plot_data_item is of type ModelDataItem.

part. It ensures that when the redraw function replaces the Spectrum1D object in this data item, the data item does not return a Spectrum1D with the wrong flux values. The `flux = self.flux` line updates the flux values of the Spectrum1D by overriding its `_data` attribute (see the `flux` property for how it does this).
